tests/chart_tests/test_private_registry.py

```diff
@@ -69,3 +71,63 @@ def test_private_registry_repository_overrides_work():
                     )
                     differently_named_images.append(image)
     assert not differently_named_images, differently_named_images
+
+
+def get_private_registry_docs_image_pull_secrets():
+    repository = "bob-the-registry"
+    secret_name = "bob-the-registry-secret"
+
+    kubernetes_objects = {
+        "Deployment": "spec.template.spec.imagePullSecrets",
+        "StatefulSet": "spec.template.spec.imagePullSecrets",
+        "Job": "spec.template.spec.imagePullSecrets",
+        "DaemonSet": "spec.template.spec.imagePullSecrets",
+        "Pod": "spec.template.spec.imagePullSecrets",
+        "CronJob": "spec.jobTemplate.spec.template.spec.imagePullSecrets",
+    }
+
+    docs = render_chart(
+        values={
+            "global": {
+                "privateRegistry": {
+                    "enabled": True,
+                    "repository": repository,
+                    "secretName": secret_name,
+                }
+            }
+        },
+    )
+
+    searched_docs = []
+    for key, val in kubernetes_objects.items():
+        searched_docs += jmespath.search(
+            "[?kind == `"
+            + key
+            + "`].{name: metadata.name, kind: kind, image_pull_secrets: "
+            + val
+            + "}",
+            docs,
+        )
```

Comment:
This one is tricky because of the brace interpretation, but I think it would be more readable if we used format strings. We could also reduce it into a comprehension. WDYT of this:

```suggestion
ss = "[?kind == `%s`].{name: metadata.name, kind: kind, image_pull_secrets: %s}"
searched_docs = [doc for k_obj in kubernetes_objects.items() for doc in jmespath.search(ss % (k_obj[0], k_obj[1]), docs)]
```